core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -6526,12 +6518,6 @@ sealed trait Exit[+E, +A] extends ZIO[Any, E, A] { self =>
   final def isSuccess: Boolean =
     self.isInstanceOf[Success[?]]
 
-  /**
-   * Returns an effect whose success is mapped by the specified `f` function.
-   */
-  override final def map[B](f: A => B)(implicit trace: Trace): ZIO[Any, E, B] =
-    mapExit(f)
-
```

Comment:
@hearnadam since you already reviewed just wanted to point out that I made this change (also with `fold`) since your last review. I think this overriding is no longer required because we already override `flatMap` to produce an exit. This allows us to make the `map` and `fold` methods `final` on `ZIO` which means the JVM can apply better optimizations / inlining as the method is no longer overwritten.

It does result in slightly more allocations in cases that `map` / `fold` is used on an `Exit`, but to be honest I think those cases not very common and we can always use `mapExit` / `foldExit` on them.